### PR TITLE
hxp template

### PR DESCRIPTION
**Pull request by buckle2000**
might be useful.
still need code support.

### CODE REVIEW

**Comment by Gama11** [comment]
Interesting, I've been meaning to check out the `.hxp` format. By "code support" you mean adjusting flixel-tools?

File: default/project.hxp.tpl [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import lime.project.*;
+
+class Project extends HXProject {
```
Comment (Gama11) [inline default/project.hxp.tpl]: Doesn't match Flixel's code style, `{` go in a new line.

File: default/project.hxp.tpl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		
+		defines.set("SWF_VERSION", "11.8");
+		
+		
+        sources.push("source");
+		
+		// put haxe library names you want to use below 
+		var libs = ["flixel", /*"flixel-addons",*/ /*"flixel-ui",*/ /*"nape"*/];
+		for (l in libs)
+			haxelibs.push (new Haxelib (l));
```
Comment (Gama11) [inline default/project.hxp.tpl]: Doesn't match Flixel's code style, no spaces between method calls and `(`.

File: default/project.hxp.tpl [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import lime.project.*;
+
+class Project extends HXProject {
+	
+	public function new() {
+		super();
+		
+        meta = { packageName: ${PROJECT_NAME}, company: "HaxeFlixel", version: "0.0.1" };
```
Comment (Gama11) [inline default/project.hxp.tpl]: Indent should only be tabs, there are spaces here and there in this file.

**Comment by buckle2000** [comment]
Yes.
at the mean time, only one of project.xml or project.hxp can exist.

**Comment by Gama11** [comment]
Maybe it should be `Project.hxp` instead of `project.hxp` for consistency with `Project.xml`?

**Comment by buckle2000** [comment]
Thanks to (Astyle)[http://astyle.sourceforge.net/].

do you mean `astyle --style=ansi --indent=tab -Y {filename}` ?

File: default/project.hxp.tpl [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		sources.push("source");
+
+		// put haxe library names you want to use below
+		var libs = ["flixel", /*"flixel-addons",*/ /*"flixel-ui",*/ /*"nape"*/];
+		for (l in libs)
+			haxelibs.push(new Haxelib (l));
+
+		includeAssets("assets");
+
+		window.width = $ {WIDTH};
```
Comment (Gama11) [inline default/project.hxp.tpl]: There's a space between the `$` and `{` here, I doubt the replacement still works with that.

**Comment by buckle2000** [comment]
Fixed.
I have no idea how this happened.
Maybe I should not believe astyle.

**Comment by Gama11** [comment]
Are you planning to adjust flixel-tools as well?

**Comment by buckle2000** [comment]
If I have time. :)

**Comment by buckle2000** [comment]
Changed Main.hx.
So the users do not have to change screen resolution both in project.xml and Main.hx.
Also, I think Main.hx should deal with screen resizing event.

**Comment by Gama11** [comment]
In what way?

**Comment by buckle2000** [comment]
Nevermind, my fault. I found [FlxGame](https://github.com/HaxeFlixel/flixel/blob/dev/flixel/FlxGame.hx#L344) already dealt with that.

**Comment by buckle2000** [comment]
Still not mature.

**Pull request closed without merging**